### PR TITLE
Support vector sets in MOI wrapper

### PR DESCRIPTION
**Pull request by blegat**
Second option of the two options listed in https://github.com/jump-dev/Convex.jl/pull/668

Closes https://github.com/jump-dev/Convex.jl/pull/668

### CODE REVIEW

File: src/MOI_wrapper.jl [hunk 1]
```diff
@@ -117,8 +117,8 @@ end
 
 function MOI.supports_constraint(
     ::Optimizer{T},
-    ::Type{MOI.ScalarNonlinearFunction},
-    ::Type{<:Union{MOI.EqualTo{T},MOI.GreaterThan{T},MOI.LessThan{T}}},
+    ::Type{MOI.VectorNonlinearFunction},
+    ::Type{<:MOI.AbstractVectorSet},
```
Comment (blegat) [inline src/MOI_wrapper.jl:121]: We should probably have a list of sets for which we define `vexity` instead because otherwise, if there is a set for which `vexity` is not defined but there is a bridge to convert it into sets for which `vexity` is defined, it won't be applied because Convex lied saying that `vexity` is defined

Comment (blegat) [reply src/MOI_wrapper.jl:121]: I added a comment

File: src/Constraint.jl [hunk 2]
```diff
@@ -3,17 +3,17 @@
 # Use of this source code is governed by a BSD-style license that can be found
 # in the LICENSE file or at https://opensource.org/license/bsd-2-clause
 
-mutable struct Constraint{S<:MOI.AbstractSet}
+mutable struct Constraint{S<:MOI.AbstractVectorSet}
```
Comment (odow) [inline src/Constraint.jl:6]: Why is this needed?

Comment (blegat) [reply src/Constraint.jl:6]: It's not needed, I just thought we are only supporting vector sets at the moment, we can revert that part

Comment (blegat) [reply src/Constraint.jl:6]: For example, isn't `populate_dual!` always returning a vector ?

Comment (odow) [reply src/Constraint.jl:6]: Ah true. We can always extend this non-breaking in future.

Comment (odow) [reply src/Constraint.jl:6]: I guess there'd be an error like this at the moment:
```julia
julia> reshape(2.0, (1,1))
ERROR: MethodError: no method matching reshape(::Float64, ::Tuple{Int64, Int64})
```

So we'd need a scalar `populate_dual!` method, but that is do-able (once there is demand).

File: src/MOI_wrapper.jl (hunk 1 above)
Comment (odow) [reply src/MOI_wrapper.jl:121]: We'll say `true`, and then end up calling a missing vexity and get ```"`Convex.vexity(vex, ::$(typeof(set)))`: is not yet implemented. Please open an issue at https://github.com/jump-dev/Convex.jl"```, so that seems like an okay error. It tells them to open an issue, at least.

**Review by odow — APPROVED** [review]
LGTM. I tweaked the comment
